packages/aws-rfdk/lib/deadline/lib/secrets-management-ref.ts

```diff
@@ -0,0 +1,106 @@
+ *
+ * See https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html#assigned-roles
+ */
+export enum SecretsManagementRole {
+  /**
+   * The administrator role is given to users that are created either by the Repository Installer when enabling the
+   * Deadline Secrets Management feature for the first time, or by running the CreateNewAdmin command. Note: there can
+   * be more than one Administrator user. All Administrators are equal and have full read and write access to all
+   * secrets.
+   */
+  ADMINISTRATOR = 'Administrator',
+
+  /**
+   * The Server role is intended to be granted to your machine(s) running the
+   * [Remote Connection Server](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/remote-connection-server.html#remote-connection-server-ref-label)
+   * application. The Server role is granted to a registered machine by an administrator in the
+   * [Monitor UI](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html#identity-management-assigning-ref-label).
+   * In order to encrypt and decrypt secrets, the master key must be assigned to the Server by and Administrator user
+   * running the [GrantKeyAccessToServer command](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html#deadline-secrets-management-command-grantkeyaccesstoserver).
+   * Servers can encrypt and decrypt all secrets, and are responsible for providing secrets to approved clients.
+   */
+  SERVER = 'Server',
+
+  /**
+   * The Client role is typically intended to be granted to any of your machines running the
+   * [Worker](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/worker.html#worker-ref-label)
+   * application. The Client role is granted to a registered machine by an administrator in the
+   * [Monitor UI](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html#identity-management-assigning-ref-label).
+   * Clients can request storage of secrets not in the
+   * [Administrator Secret Access Level](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html#deadline-secrets-management-secret-namespace-ref-label),
+   * and can retrieve secrets from all namespaces when authenticating through the server.
+   */
+  CLIENT = 'Client',
+};
+
+/**
+ * The different possible Deadline Secrets Management registration statuses that a Deadline Client's identity can be set
+ * to.
+ *
+ * See https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html#registration-status
+ */
+export enum SecretsManagementRegistrationStatus {
```

Comment:
What the point to have these statuses? Do we expect users will use something except `REGISTERED`?

Reply:
I made all possible identity registration statuses available that Deadline supports. RFDK users should not need to interact with this part of the API, since the constructs will automatically work together to configure the identity registration settings.

The only way a user could use this API is if they called `RenderQueue.configureSecretsManagementAutoRegistration(...)` directly. On that note, I'll document that fact on the method's TSDoc header.

Reply:
I mean, since we do not need others statuses we can just use constant `REGISTERED` in python script and do not ask for status in `SubnetIdentityRegistrationSettingsProps` unless we want to register settings with different statuses.